src/Service/MolliePaymentMailService.php

```diff
@@ -0,0 +1,210 @@
+            'success' => true,
+            'message' => $this->module->l('Second chance email was successfully send!'),
+            'checkoutUrl' => $checkoutUrl
+        ];
+        }
+        $qrCode = false;
+
+        $cart = new Cart($paymentApi->metadata->cart_id);
+        $customer = new Customer($cart->id_customer);
+        $paymentData = [
+            'amount' => [
+                'value' => $paymentApi->amount->value,
+                'currency' => $paymentApi->amount->currency,
+            ],
+            'redirectUrl' =>($qrCode
+                ? $context->link->getModuleLink(
+                    'mollie',
+                    'qrcode',
+                    ['cart_id' => $paymentApi->metadata->cart_id, 'done' => 1, 'rand' => time()],
+                    true
+                )
+                : $context->link->getModuleLink(
+                    'mollie',
+                    'return',
+                    [
+                        'cart_id' => $paymentApi->metadata->cart_id,
+                        'utm_nooverride' => 1,
+                        'rand' => time(),
+                        'key' => $customer->secure_key,
+                        'customerId' => $customer->id
+                    ],
+                    true
+                )
+            ),
+            'description' => $paymentApi->description,
+            'metadata' => [
+                'cart_id' => $paymentApi->metadata->cart_id,
+                'order_reference' => $paymentApi->metadata->order_reference,
+                'secure_key' => $paymentApi->metadata->secure_key
+            ],
+        ];
+
+        if (!EnvironmentUtility::isLocalEnvironment()) {
+            $paymentData['webhookUrl'] = $context->link->getModuleLink(
+                'mollie',
+                'webhook',
+                [],
+                true
+            );
+        }
+        $newPayment = $api->payments->create($paymentData);
+
+        if(isset($newPayment)) {
+            $updateTransactionId = $this->paymentMethodRepository->updateTransactionId($transactionId, $newPayment->id);
+
+            if($updateTransactionId) {
+                $checkoutUrl = $newPayment->getCheckoutUrl();
+                return [
+                    'success' => true,
+                    'message' => $this->module->l('Second chance email was successfully send!'),
+                    'checkoutUrl' => $checkoutUrl
+                ];
+            }
+        }
+
+        return
+            [
+                'success' => false,
+                'message' =>
+                    $this->module->l('Failed to send second chance email!')
+            ];
+    }
+
+    private function getCheckoutUrl($molliePayments)
+    {
+        $checkoutUrl = '';
+        /** @var Payment $molliePayment */
+        foreach ($molliePayments as $molliePayment) {
+            if ($molliePayment->status === PaymentStatus::STATUS_OPEN) {
+                $checkoutUrl = $molliePayment->getCheckoutUrl();
```

Comment:
couldn't you break the loop after getting the first url? Unless it is supposed to loop through all of them and get the last one